### ISSUE TITLE
Question on Cosmos DB database provider owned object querying null

### ISSUE THREAD

**Issue by codemonkeh-dave** [dotnet/efcore#24087]
## Question

I am trying to query against owned objects to see if they are null but I get an exception and can't find a work-around to achieving the query (see sample code).

I know I can add `.AsEnumerable()` but this will make it a client-side query which I don' t want as it will surely bring all records from the database into process to run the query/filter.

If this is clearly documented somewhere I'm sorry for missing it.

If it makes a difference to the answer my production container is used for multiple types and so I use .HasDiscriminator().

### Include your code

To demonstrate the issue, I am using the sample found here https://github.com/dotnet/EntityFramework.Docs/tree/master/samples/core/Cosmos and have added the below simple query into sample.cs.

```C#
using (var context = new OrderContext())
{
    var ordersWithAnAddress = await context.Orders.Where(o => o.ShippingAddress != null).ToListAsync();
}
```

### Error
```
System.InvalidOperationException: 'The LINQ expression 'DbSet<Order>()
    .Where(o => EF.Property<StreetAddress>(o, "ShippingAddress") != null)' could not be translated. Additional information: Translation of member 'OrderId' on entity type 'Order.ShippingAddress#StreetAddress' failed. This commonly occurs when the specified member is unmapped.
Translation of member 'OrderPartitionKey' on entity type 'Order.ShippingAddress#StreetAddress' failed. This commonly occurs when the specified member is unmapped. Either rewrite the query in a form that can be translated, or switch to client evaluation explicitly by inserting a call to 'AsEnumerable', 'AsAsyncEnumerable', 'ToList', or 'ToListAsync'. See https://go.microsoft.com/fwlink/?linkid=2101038 for more information.'
```
### Provider and version information

EF Core version: 5.0.2
Database provider: Microsoft.EntityFrameworkCore.Cosmos (version 5.0.2)
Target framework: This example is .NET 5.0 but the issue also happens in my project in .netcore 3.2
Operating system: Windows 10
IDE: (e.g. Visual Studio 2019 16.8.4)

**Comment by smitpatel** [comment]
Blocked on #17722

The issue happens since we try to rewrite to entity equality but the property doesn't exist on server side for owned entity. We need to translate it to `IS_DEFINED` which requires refactoring in cosmos SQL tree.

**Comment by nlpMbordogna** [comment]
I see you guys are punting on this.   This is kind of an importing thing, we can't safely query any owned entities without it.  Do you have a work around? we are dealing with the IQueryable

**Comment by gabolarraguivel** [comment]
Any update on this? Is there any workaround?

**Comment by aladd04** [comment]
Any chances this gets looked at again, or has anyone come across a workaround?

**Comment by JoasE** [comment]
I have a PR open for this with a possible solution in #37321

**Comment by JoasE** [comment]
I found these issues which I think are related: #17746 #13131 and read up on them a bit.

**What I think needs to happen is for my PR (#37321) to implement** ( `c => c.Prop1.Prop2 = null` becomes `WHERE c["Prop1"]["Prop2"] = null`) **and then add a separate PR for #17746 expanding nested null checks** (`c.Prop1.Prop2 = null` becomes `c["Prop1"] = null OR c["Prop1"]["Prop2"] = null`). **And I would like to confirm whether that is correct**

From reading the issues: The conclusion seems to be that the EF query pipeline should only treat `null` (json) as `null` (C#), and `undefined` should be handled separately by the user. You must use `= null` syntax for this in cosmos db as `NOT IS_NULL(c.Prop1)` will be true when `c.Prop1` is `undefined`.
However, the query pipeline should treat a property being `undefined` because of a parent property being `null` as `null`. (#17746) This can be handled in cosmos by including a null check at every level. (`c.Prop1.Prop2 = null` becomes `c["Prop1"] = null OR c["Prop1"]["Prop2"] = null`)

What confuses me a bit was why this was blocked on #17722 if the above is correct, as it wouldn't be needed. Maybe that was done by mistake?

My current PR, started of with translating `c => c.Prop1.Prop2 = null` to `WHERE c["Prop1"]["Prop2"] = null`, which would only match if `Prop1` had a value for `Prop2` of null. but I moved to `WHERE IS_NULL(c["Prop1"]["Prop2"]) OR NOT IS_DEFINED(c["Prop1"]["Prop2"])` later, which would match if `Prop1` or `Prop2` is either `null` or `undefined`. From what I am gathering, the latter isn't the intention. Current pr also adds tests for nested null scenario's, which I would skip/remove for now until nested null checks are implemented.

@roji Side note that I would like to use the outcome of this for the query pipeline of #37577 (which is almost ready to be published) as there are some conflicts with entity equality

**Comment by roji** [comment]
@JoasE [see my comment](https://github.com/dotnet/efcore/issues/17746#issuecomment-3829345798) on #17746 on the `null` and `undefined` question. So I think we should indeed allow comparison of JSON objects to null (as you're doing in #37321 for owned JSON), but not do any sort of expansion checks for nested properties, or involve special handling for `undefined`.

However, that's purely my view and I'm definitely interesting in having a discussion if you (or anyone else) thinks we should go another way.

Regardless, I've given your #37321 a review.
